lldb/tools/lldb-dap/Protocol.h

```diff
@@ -141,15 +137,81 @@ bool fromJSON(const llvm::json::Value &, Event &, llvm::json::Path);
 //   ]
 // }
 struct Response {
+  enum class Message {
+    cancelled,
+    notStopped,
+  };
+
   int64_t request_seq;
-  bool success;
   std::string command;
-  std::optional<std::string> message;
+  bool success;
+  std::optional<Message> message;
   std::optional<llvm::json::Value> rawBody;
 };
 bool fromJSON(const llvm::json::Value &, Response &, llvm::json::Path);
 llvm::json::Value toJSON(const Response &);
 
+// "Message": {
+//   "type": "object",
+//   "description": "A structured message object. Used to return errors from
+//   requests.", "properties": {
+//     "id": {
+//       "type": "integer",
+//       "description": "Unique (within a debug adapter implementation)
+//       identifier for the message. The purpose of these error IDs is to help
+//       extension authors that have the requirement that every user visible
+//       error message needs a corresponding error number, so that users or
+//       customer support can find information about the specific error more
+//       easily."
+//     },
+//     "format": {
+//       "type": "string",
+//       "description": "A format string for the message. Embedded variables
+//       have the form `{name}`.\nIf variable name starts with an underscore
+//       character, the variable does not contain user data (PII) and can be
+//       safely used for telemetry purposes."
+//     },
+//     "variables": {
+//       "type": "object",
+//       "description": "An object used as a dictionary for looking up the
+//       variables in the format string.", "additionalProperties": {
+//         "type": "string",
+//         "description": "All dictionary values must be strings."
+//       }
+//     },
+//     "sendTelemetry": {
+//       "type": "boolean",
+//       "description": "If true send to telemetry."
+//     },
+//     "showUser": {
+//       "type": "boolean",
+//       "description": "If true show user."
+//     },
+//     "url": {
+//       "type": "string",
+//       "description": "A url where additional information about this message
+//       can be found."
+//     },
+//     "urlLabel": {
+//       "type": "string",
+//       "description": "A label that is presented to the user as the UI for
+//       opening the url."
+//     }
+//   },
+//   "required": [ "id", "format" ]
+// },
+struct ErrorMessage {
+  uint64_t id;
+  std::string format;
+  std::optional<std::map<std::string, std::string>> variables;
+  std::optional<bool> sendTelemetry;
+  std::optional<bool> showUser;
```

Comment:
afaict, there is no difference between "false" and "not set". Should we use a non-optional bool instead?

Reply:
Done.